model/standalone_driver/src/icon4py/model/standalone_driver/driver_utils.py

```diff
@@ -0,0 +1,400 @@
+    _add_dimension(dims.CellDim)
+
+    return decomposition_info
+
+
+def create_vertical_grid(
+    vertical_grid_config: v_grid.VerticalGridConfig,
+    backend: gtx_typing.Backend,
+) -> v_grid.VerticalGrid:
+    vct_a, vct_b = v_grid.get_vct_a_and_vct_b(
+        vertical_config=vertical_grid_config, allocator=backend
+    )
+
+    vertical_grid = v_grid.VerticalGrid(
+        config=vertical_grid_config,
+        vct_a=vct_a,
+        vct_b=vct_b,
+    )
+    return vertical_grid
+
+
+def create_static_field_factories(
+    grid_manager: gm.GridManager,
+    decomposition_info: decomposition_defs.DecompositionInfo,
+    vertical_grid: v_grid.VerticalGrid,
+    cell_topography: data_alloc.NDArray,
+    backend: gtx_typing.Backend,
+) -> driver_states.StaticFieldFactories:
+    geometry_field_source = grid_geometry.GridGeometry(
+        grid=grid_manager.grid,
+        decomposition_info=decomposition_info,
+        backend=backend,
+        coordinates=grid_manager.coordinates,
+        extra_fields=grid_manager.geometry_fields,
+        metadata=geometry_meta.attrs,
+    )
+
+    interpolation_field_source = interpolation_factory.InterpolationFieldsFactory(
+        grid=grid_manager.grid,
+        decomposition_info=decomposition_info,
+        geometry_source=geometry_field_source,
+        backend=backend,
+        metadata=interpolation_attributes.attrs,
+    )
+
+    metrics_field_source = metrics_factory.MetricsFieldsFactory(
+        grid=grid_manager.grid,
+        vertical_grid=vertical_grid,
+        decomposition_info=decomposition_info,
+        geometry_source=geometry_field_source,
+        topography=gtx.as_field((dims.CellDim,), data=cell_topography, allocator=backend),
+        interpolation_source=interpolation_field_source,
+        backend=backend,
+        metadata=metrics_attributes.attrs,
+        rayleigh_type=RayleighType.KLEMP,
+        rayleigh_coeff=0.1,
+        exner_expol=0.333,
+        vwind_offctr=0.2,
+    )
+
+    return driver_states.StaticFieldFactories(
+        geometry_field_source, interpolation_field_source, metrics_field_source
+    )
+
+
+def initialize_granule(
+    grid: icon_grid.IconGrid,
+    parallel_props: decomposition_defs.ProcessProperties,
+    decomposition_info: decomposition_defs.DecompositionInfo,
+    vertical_grid: v_grid.VerticalGrid,
+    diffusion_config: diffusion.DiffusionConfig,
+    solve_nh_config: solve_nh.NonHydrostaticConfig,
+    static_field_factories: driver_states.StaticFieldFactories,
+    backend: gtx_typing.Backend | None,
+) -> tuple[
+    diffusion.Diffusion,
+    solve_nh.SolveNonhydro,
+]:
+    exchange = decomposition_defs.create_exchange(parallel_props, decomposition_info)
+
+    geometry_field_source = static_field_factories.geometry_field_source
+    interpolation_field_source = static_field_factories.interpolation_field_source
+    metrics_field_source = static_field_factories.metrics_field_source
+
+    log.info("Start creating cell geometry")
```

Comment:
```suggestion
    log.info("creating cell geometry")
```
This and others below to be consistent with the logs from `standalone_driver.py`